### PR TITLE
Not wrap IOException twice form checkpoint

### PR DESCRIPTION
**Pull request by wenbingshen**
### Motivation

IOException may be wrapped by RuntimeException, so don't wrap RuntimeException with IOException.

### CODE REVIEW

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java [hunk 1]
```diff
@@ -839,6 +839,10 @@ public void checkpoint(Checkpoint checkpoint) throws IOException {
             throw e;
         } catch (RuntimeException e) {
             recordFailedEvent(dbLedgerStorageStats.getFlushStats(), startTime);
+            // not wrap IOException again
+            if (e.getCause() != null && e.getCause() instanceof IOException) {
```
Comment (gaozhangmin) [inline bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java:843]: Does IOException throw at the first catch block?

Comment (wenbingshen) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java:843]: This IOException is wrapped by RuntimeException. not a directly IOException. The first catch block can not caught it.

Comment (hangc0276) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java:843]: Could you share the detailed IOException wrapped by RuntimeException?

**Review by hangc0276 — CHANGES_REQUESTED** [review]

File: bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java (hunk 1 above)
Comment (horizonzy) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java:843]: Change the interface will be better:
```
    public interface EntryConsumer {
        void accept(long ledgerId, long entryId, ByteBuf entry);
    }
```

```
    public interface EntryConsumer {
        void accept(long ledgerId, long entryId, ByteBuf entry) throws IOException;
    }
```
So we needn't catch the inner exception. Throw the IOException directly.

Comment (hangc0276) [reply bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/storage/ldb/SingleDirectoryDbLedgerStorage.java:843]: +1

**Review by horizonzy — APPROVED** [review]
LGTM.

**Comment by hangc0276** [comment]
@wenbingshen Would you please resolve the conflicts? thanks a lot.

**Review by hangc0276 — APPROVED** [review]

**Pull request merged**